lib/mini_twitter_client.rb

```diff
@@ -0,0 +1,81 @@
+require 'faraday'
+require 'json'
+
+class Tweet
+  def initialize(name = 'name', email = 'email@email.com', message = 'message', id = nil)
+    @name = name
+    @email = email
+    @message = message
+    @id =id
+  end
+  def name
+    @name
+  end
+  def email
+    @email
+  end
+  def message
+    @message
+  end
+  def id
+    @id
+  end
+end
+
+class MiniTwitterClient
+  def initialize
+    @connection = Faraday.new(
+      url: 'http://localhost:4000',
+      params: {},
+      headers: {
+        'Accept' => 'application/json',
+        'Content-Type' => 'application/json'
+      }
+    )
+  end
+
+  def get_tweets
+    res = @connection.get('/api/tweets')
+    data = JSON.parse(res.body)['data'].map do |tweet|
+      Tweet.new(tweet['author']['name'], tweet['author']['email'], tweet['message'], tweet['id'])
+    end
+
+    {
+      status: res.status.to_i,
+      data:
+    }
+  end
+
+  def create_tweet(name, email, message)
+    res = @connection.post('/api/tweets') do |post|
+      post.body = { "tweet": { "author": { "name": name, "email": email }, "message": message } }.to_json
+    end
```

Comment:
Personally, I would write it this way:

```suggestion
    body = {
      "tweet": {
        "author": {
          "name": name,
          "email": email
        },
        "message": message
      }
    }.to_json
    res = @connection.post('/api/tweets', body)
```

But, it may be just a personal preference ;)

You can see in the documentation that `post` method accepts `body` as a second argument: https://www.rubydoc.info/gems/faraday/Faraday%2FConnection.post.

Reply:
Much better than my version